modules/apps/bookmarks/bookmarks-web/src/main/java/com/liferay/bookmarks/web/internal/info/display/contributor/BookmarksEntryInfoDisplayContributor.java

```diff
@@ -0,0 +1,127 @@
+import com.liferay.asset.kernel.model.AssetEntry;
+import com.liferay.bookmarks.model.BookmarksEntry;
+import com.liferay.bookmarks.service.BookmarksEntryService;
+import com.liferay.info.display.contributor.InfoDisplayContributor;
+import com.liferay.info.display.contributor.InfoDisplayField;
+import com.liferay.info.display.contributor.InfoDisplayObjectProvider;
+import com.liferay.info.display.field.ExpandoInfoDisplayFieldProvider;
+import com.liferay.info.display.field.InfoDisplayFieldProvider;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yang Cao
+ */
+@Component(enabled = false, service = InfoDisplayContributor.class)
+public class BookmarksEntryInfoDisplayContributor
+	implements InfoDisplayContributor<BookmarksEntry> {
+
+	@Override
+	public String getClassName() {
+		return BookmarksEntry.class.getName();
+	}
+
+	@Override
+	public Set<InfoDisplayField> getInfoDisplayFields(
+			long classTypeId, Locale locale)
+		throws PortalException {
+
+		Set<InfoDisplayField> infoDisplayFields =
+			_infoDisplayFieldProvider.getContributorInfoDisplayFields(
+				locale, AssetEntry.class.getName(),
+				BookmarksEntry.class.getName());
+
+		infoDisplayFields.addAll(
+			_expandoInfoDisplayFieldProvider.
+				getContributorExpandoInfoDisplayFields(
+					BookmarksEntry.class.getName(), locale));
+
+		return infoDisplayFields;
+	}
+
+	@Override
+	public Map<String, Object> getInfoDisplayFieldsValues(
+			BookmarksEntry bookmarksEntry, Locale locale)
+		throws PortalException {
+
+		return HashMapBuilder.<String, Object>putAll(
+			_assetEntryInfoDisplayFieldProvider.
+				getAssetEntryInfoDisplayFieldsValues(
+					BookmarksEntry.class.getName(), bookmarksEntry.getEntryId(),
+					locale)
+		).putAll(
+			_expandoInfoDisplayFieldProvider.
+				getContributorExpandoInfoDisplayFieldsValues(
+					BookmarksEntry.class.getName(), bookmarksEntry, locale)
+		).putAll(
+			_infoDisplayFieldProvider.getContributorInfoDisplayFieldsValues(
+				BookmarksEntry.class.getName(), bookmarksEntry, locale)
+		).build();
+	}
+
+	@Override
+	public InfoDisplayObjectProvider<BookmarksEntry>
+			getInfoDisplayObjectProvider(long classPK)
+		throws PortalException {
+
+		BookmarksEntry bookmarksEntry = _bookmarksEntryService.getEntry(
+			classPK);
+
+		if (bookmarksEntry.isInTrash()) {
+			return null;
+		}
+
+		return new BookmarksInfoDisplayObjectProvider(bookmarksEntry);
+	}
+
+	@Override
+	public InfoDisplayObjectProvider<BookmarksEntry>
+			getInfoDisplayObjectProvider(long groupId, String urlTitle)
+		throws PortalException {
+
+		return getInfoDisplayObjectProvider(Long.valueOf(urlTitle));
+	}
+
+	@Override
+	public String getInfoURLSeparator() {
+		return "/b/";
```

Comment:
You cannot use `/b` here; it is already in use by blogs. See `FriendlyURLResolverConstants` class for the set of separators in use.